Packs/ThalesCipherTrustManager/Integrations/ThalesCipherTrustManager/ThalesCipherTrustManager.py

```diff
@@ -0,0 +1,919 @@
+        )
+
+    def delete_external_certificate(self, external_cert_id: str) -> dict[str, Any]:
+        return self._http_request(
+            method='DELETE',
+            url_suffix=urljoin(EXTERNAL_CAS_URL_SUFFIX, external_cert_id),
+            return_empty_response=True,
+        )
+
+    def update_external_certificate(self, external_ca_id: str, request_data: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='PATCH',
+            url_suffix=urljoin(EXTERNAL_CAS_URL_SUFFIX, external_ca_id),
+            json_data=request_data,
+        )
+
+    def get_external_certificates_list(self, params: dict) -> dict[str, Any]:
+        return self._http_request(
+            method='GET',
+            url_suffix=EXTERNAL_CAS_URL_SUFFIX,
+            params=params,
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def derive_skip_and_limit_for_pagination(limit_str: Optional[str], page_str: Optional[str], page_size_str: Optional[str]) -> \
+        tuple[int, int]:
+    if page_str is not None:
+        page_from_arg = arg_to_number(page_str)
+        if page_from_arg is None:
+            raise ValueError(f'Invalid page number: {page_str}')
+        page_size_from_arg = arg_to_number(page_size_str)
+        page_size: int = page_size_from_arg if page_size_from_arg is not None else DEFAULT_PAGE_SIZE
+        if page_size > MAX_PAGE_SIZE:
+            raise ValueError(f'Page size cannot exceed {MAX_PAGE_SIZE}')
+        return (int(page_from_arg) - 1) * page_size, page_size
+    limit_from_arg = arg_to_number(limit_str)
+    limit: int = limit_from_arg if limit_from_arg is not None else DEFAULT_LIMIT
+    return 0, limit
+
+
+def optional_arg_to_bool(arg: Optional[str]) -> Optional[bool]:
+    return argToBoolean(arg) if arg is not None else arg
+
+
+def optional_arg_to_datetime_string(arg: Optional[str], date_format: str = DATE_FORMAT) -> Optional[str]:
+    datetime_object = arg_to_datetime(arg)
+    return datetime_object.strftime(date_format) if datetime_object is not None else datetime_object
+
+
+def add_empty_date_param(request_data: dict, argument_value: Optional[str], param_name: str,
+                         empty_arg_value: str = "empty") -> None:
+    if argument_value is not None:
+        request_data[param_name] = "" if argument_value == empty_arg_value else optional_arg_to_datetime_string(argument_value)
+
+
+def add_empty_list_param(request_data: dict, argument_value: Optional[str], param_name: str,
+                         empty_arg_value: str = "empty") -> None:
+    if argument_value is not None:
+        request_data[param_name] = [] if argument_value == empty_arg_value else argToList(argument_value)
+
+
+def add_login_flags(request_data: dict, argument_value: Optional[bool], flag_name: str) -> None:
+    if argument_value is not None:
+        if 'login_flags' not in request_data:
+            request_data['login_flags'] = {}
+        request_data['login_flags'][flag_name] = argument_value
```

Comment:
- Argument value is boolean thus we need to check is None rather than if not. 
- login_flags need to be manually added to match the required api format: 
![image](https://github.com/demisto/content/assets/94686128/7ed65edf-60d7-4e47-9c59-5ca6f3789000)
(The design document instructs to pass the prevent_ui_login flag rather than the login_flags object)